ext/mysqli/mysqli.c

```diff
@@ -320,9 +322,14 @@ zval *mysqli_read_property(zval *object, zval *member, int type, void **cache_sl
 	}
 
 	if (hnd) {
-		retval = hnd->read_func(obj, rv);
-		if (retval == NULL) {
-			retval = &EG(uninitialized_zval);
+		if (hnd->read_func(obj, rv, type == BP_VAR_IS) == SUCCESS && rv != NULL) {
```

Comment:
I tried to fix the failing test case by changing how property reading works, but it didn't work, so unfortunately I'm out of ideas. :(

Reply:
We need to be careful about the convention here: If the callback returns `FAILURE`, is `rv` still expected to be valid or not?

I would say that we should just make this code

```
if (hnd->read_func(obj, rv, type == BP_VAR_IS) == SUCCESS) {
    retval = rv;
} else {
    retval = &EG(uninitialized_zval);
}
```

and stop writing to the return value if we return FAILURE. This might change some false to null, but I don't think that is particularly important, especially as this is going to throw soon anyway.

Reply:
Good idea! However, we have some tests (e.g. mysqli_class_mysqli_properties_no_conn.phpt) that checks if the properties have the same value as the procedural functions: `assert(@mysqli_connect_error() === @$mysqli->connect_error);`.

By changing default `false` values of properties to `null`, these are no longer true.  Isn't it a problem?

Reply:
Hm, yeah, that might be problematic. In that case I'd suggest to go with

```
if (hnd->read_func(obj, rv, type == BP_VAR_IS) == SUCCESS || type != BP_VAR_IS) {
    retval = rv;
} else {
    retval = &EG(uninitialized_zval);
}
```

and make sure we always initialize retval even if we return FAILURE (in particular in the warning code).